main/src/com/google/refine/expr/functions/Cross.java

```diff
@@ -56,10 +56,8 @@ public Object call(Properties bindings, Object[] args) {
             long targetProjectID;
             ProjectLookup lookup;
 
-            if (v != null &&
-                    (v instanceof String || v instanceof WrappedCell) &&
-                    targetProjectName != null && targetProjectName instanceof String &&
-                    targetColumnName != null && targetColumnName instanceof String) {
+            if ((v instanceof String || v instanceof Long || v instanceof Integer || v instanceof WrappedCell)
```

Comment:
Yeah, I'll make the change.

Reply:
Do you think we should also support date? It's possible that we just require the value not to be null here.

Reply:
Basically, we can support any type as long as it's valid type of a cell value.

Reply:
Correct, any valid cell type.

History:
What we traditionally did was try to make `cross()` a bit more rigid so that users didn't have to experience as much "Ugh! My Cross is not working for some unknown reason".

Today:
And what we have found out over time is that lookup failures are OK for users to experience, because there are lots of reasons why a `cross()` might fail to perform the lookup correctly due to the data quality.  For instance, any one of [these kinds of errors](https://howtovlookupinexcel.com/13-common-problems-with-vlookups/).  So, I think we can relax the `cross()` completely as you suggest and have less defensive coding here for `cross()`.

Future:
Instead, we should ensure that we have much better error output for the failures of `cross()`.  For instance, when applying a GREL expression, users have a radio button option to `store error` and that's where we could have `cross()` produce much more informative errors about the cross lookup failure similar to how other data tools work as seen in that hyperlink provided.

Reply:
> Correct, any valid cell type.

I'm also for anytype. It will give us so much possibilities, now that we can call it in Transform and that we can call it from any rows. 

For my, a `cross(false, "TEST-Project", "isActive")` would give me the list of all rows that are active… 

And, why not, a `forEach(cross(false, "TEST-Project", "isActive"), vRow, if(vRow.cells["isAdmin"], vRow.cells["FullName"], "")).join(", ")` or so, wouldn't it be a sort a cross join? (more a single join with filtering… but it could be a way to use it.

> Future:
> Instead, we should ensure that we have much better error output for the failures of `cross()`. For instance, when applying a GREL expression, users have a radio button option to `store error` and that's where we could have `cross()` produce much more informative errors about the cross lookup failure similar to how other data tools work as seen in that hyperlink provided. With that idea, I could see where users might want an enhancement for `cross()` to have a new boolean option for "exact match with data type"

More precise error reporting is always good and can save a lot of time… Good for user experience! ;-)

I really like the way these discussion are going. Lot of interesting avenues!

Regards, Antoine